proxy/HostStatus.h

```diff
@@ -168,6 +175,35 @@ struct HostStatRec {
   }
 };
 
+struct HostStatusSync;
+typedef int (HostStatusSync::*HostStatusSyncHandler)(int, void *);
+struct HostStatusSync : public Continuation {
+  std::string hostRecordsFile;
+
+  void sync_task();
+  void
+  getHostStatusPersistentFilePath()
+  {
+    std::string rundir(RecConfigReadRuntimeDir());
+    hostRecordsFile = Layout::relative_to(rundir, ts::filename::HOST_RECORDS);
+  }
+
+public:
+  HostStatusSync()
+  {
+    getHostStatusPersistentFilePath();
+
+    SET_HANDLER((HostStatusSyncHandler)&HostStatusSync::mainEvent);
```

Comment:
You should not need to cast the type of the event function anymore because of https://github.com/apache/trafficserver/pull/6996 .

It is best to not use C casts in new code.  A C cast is equivalent to reinterpret_cast and is hard to find with grep.

Reply:
@ywkaras ok, it's removed.